### ISSUE TITLE
RUF037 false negative on `deque(f"{""}")`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#20050]
### Summary

[`unnecessary-empty-iterable-within-deque-call` (RUF037)](https://docs.astral.sh/ruff/rules/unnecessary-empty-iterable-within-deque-call/) checks whether an f-string is empty in the sense of being the empty literal `f""`, whereas it should should check whether it evaluates to an empty iterable. That is, it should be implemented in terms of `is_empty_f_string` instead of `is_empty_literal`. Example:
```console
$ cat >ruf037.py <<'# EOF'
from collections import deque
deque(f"{""}")
# EOF

$ ruff --isolated check ruf037.py --select RUF037 --preview
All checks passed!
```

### Version

ruff 0.12.10 (c68ff8d90 2025-08-21)
